db/database.go

```diff
@@ -840,49 +840,73 @@ type principalsViewRow struct {
 // Returns the IDs of all users and roles
 func (db *DatabaseContext) AllPrincipalIDs() (users, roles []string, err error) {
 
-	results, err := db.QueryPrincipals()
-	if err != nil {
-		return nil, nil, err
-	}
+	startKey := ""
+	limit := 2
```

Comment:
As discussed, remove, and defensive coding to make sure limit > 1

Reply:
Defence is on config side